test/fake_lang/non_jl_test.jl

```diff
@@ -0,0 +1,44 @@
+using Pkg
+Pkg.activate(".")
+Pkg.instantiate()
+
+struct MyFile
+    file::String
+end
+Base.abspath(file::MyFile) = MyFile(Base.abspath(file.file))
+Base.isabspath(file::MyFile) = Base.isabspath(file.file)
+Base.joinpath(str::String, file::MyFile) = Myfile(Base.joinpath(str, file.file))
+Base.normpath(file::MyFile) = Myfile(Base.normpath(file.file))
+Base.isfile(file::MyFile) = Base.isfile(file.file)
+Base.findfirst(str::String, file::MyFile) = Base.findfirst(str, file.file)
+Base.String(file::MyFile) = file.file
+
+function make_module(file::MyFile)
+    exprs = []
+    for line in eachline(file.file)
+       val, name = split(line, '=')
+       push!(exprs, :(function $(Symbol(name))() $val end))
+    end
+    @show exprs
+    Expr(:toplevel, :(baremodule test
+       $(exprs...)
+    end), :(using .test))
+end
+
+function Base.include(mod::Module, file::MyFile)
+    Core.eval(mod, make_module(file))
+end
+Base.include(file::MyFile) = Base.include(Core.Main, file)
+
+using Revise
+function Revise.parse_source!(mod_exprs_sigs::Revise.ModuleExprsSigs, file::MyFile, mod::Module; kwargs...)
+    ex = make_module(file)
+    @show mod_exprs_sigs, file, mod, kwargs
+    Revise.process_source!(mod_exprs_sigs, ex, file, mod; kwargs...)
+end
+m=MyFile("test.program")
+includet(m)
+@test test.y() == "2"
+@test test.x() == "1"
+
```

Comment:
@timholy how can I edit the file from the tests?

Reply:
@KristofferC any suggestions?

Reply:
The way Revise tests work in general is to write a little mini-package, `using` it, sleep for a bit to allow the `mtime` to age a bit, and then overwrite the file with new contents. That doesn't seem to fit this particular test, but is there any reason you couldn't create a test that does depend in some way on the content of a file?

Reply:
My goal here was to add a test for `@test test.x() == "2"` after modifying the file, but I couldn't figure out a good way to modify the file.